temporalio/contrib/openai_agents/_heartbeat_decorator.py

```diff
@@ -0,0 +1,37 @@
+import asyncio
+from functools import wraps
+from typing import Any, Awaitable, Callable, TypeVar, cast
+
+from temporalio import activity
+
+F = TypeVar("F", bound=Callable[..., Awaitable[Any]])
+
+
+def _auto_heartbeater(fn: F) -> F:
+    # We want to ensure that the type hints from the original callable are
+    # available via our wrapper, so we use the functools wraps decorator
```

Comment:
Nit

```suggestion
Propagate type hints from the original callable
```

This is required for deserialization right? If so then good to mention that in the comment.